### DISCUSSION TITLE
How to customize GeoJSON lines / poly-lines

### DISCUSSION THREAD

**Discussion by iacisme** [opengeos/leafmap#328] [Q&A] [answered]
Hi;

I have a question regarding the customization of lines/polylines loaded from `GeoJSON` files.

The following map was created using `folium`:

![image](https://user-images.githubusercontent.com/72644814/206829535-91f3a0c0-ab0d-40ec-99d8-84bd1b71ae2d.png)

The red and blue pipelines in the image were generated using these `folium` scripts:

# Loading Nord Stream 1 Route

```
# Load Nord Stream 1 pipeline routing into memory
nord_stream_1_geojson_file_path = './map_data/nordstream_1_full.geojson'

# Add the routing to the map
nord_stream_1_route = folium.GeoJson(data = nord_stream_1_geojson_file_path,
                                     tooltip = "Nord Stream 1",
                                     name = "Nord Stream 1",
                                     style_function = lambda x: {'fillColor' : 'dardkblue', 'color' : 'darkblue', 'weight' : 5}
                                    ).add_to(Map);

```
# Loading Nord Stream 2 Route

```
# Load Nord Stream 2 pipeline routing into memory
nord_stream_2_geojson_file_path = './map_data/nordstream_2_full.geojson'

# Add the routing to the map
nord_stream_2_route = folium.GeoJson(data = nord_stream_2_geojson_file_path, 
                                     tooltip = "Nord Stream 2",
                                     name = "Nord Stream 2",
                                     style_function = lambda x: {'fillColor' : 'red', 'color' : 'red', 'weight' : 3}
                                    ).add_to(Map);

```

Thus allowing someone to uniquely identify one line from another.

Is there a way to customize these kinds of GeoJSON files in `leafmap` or `ipyleaflet`? I've seen the following example in the [Key Features](https://leafmap.org/notebooks/00_key_features/#add-geojson):

![image](https://user-images.githubusercontent.com/72644814/206829771-57f4790c-72dc-499f-ad2d-39f407f33506.png)

But that's not what I'm looking to do. I need the ability to customize each line.

I've tried experimenting some, based on the examples, but the results look like this:

![image](https://user-images.githubusercontent.com/72644814/206829820-2d058b45-b6f6-4592-814e-47fb5ca87dcc.png)

Here is the code I used to load the `GeoJSON` file:

# Loading Nord Stream 1 Route

```
# Load Nord Stream 1 pipeline routing into memory
nord_stream_1_geojson_file_path = './map_data/nordstream_1_full.geojson'

# Customize the pipeline color
nord_stream_1_style = {"stroke": False,
                       "color": "dardkblue",
                       "weight": 3,
                       "opacity": 1,
                       "fill": True,
                       "fillColor": "darkblue",
                       "fillOpacity": 1,
                       }

# Add UK border to the map
base_map.add_geojson(nord_stream_1_geojson_file_path,
                     layer_name = 'Nord Stream 1',
                     style = nord_stream_1_style,
                     info_mode = False,
                     )


base_map
```

For my applications (engineering) - it's going to be important for me to use GeoJSON files that represent pipelines, powerlines, and electrical cables. I'm therefore going to need the ability to customize colors and other values in order to identify one line from another.

Can someone please show me how to do this using `leafmap` or `ipyleaflet`?

Thank you.

--Igor

**Comment by iacisme** [comment, marked as answer]
Hey guys, I figured it our. You pass a basic dictionary to the `style` parameter. Here is my code:

# Loading Nord Stream 1 Route
```
# Load Nord Stream 1 pipeline routing into memory
nord_stream_1_geojson_file_path = './map_data/nordstream_1_full.geojson'

# Add pipeline route to the map
base_map.add_geojson(nord_stream_1_geojson_file_path,
                     layer_name = 'Nord Stream 1',
                     style = {"color" : "darkblue",
                              "weight" : 4,
                               },
                     info_mode = False,
                    )
```
# Loading Nord Stream 2 Route
```
# Load Nord Stream 2 pipeline routing into memory
nord_stream_2_geojson_file_path = './map_data/nordstream_2_full.geojson'

# Add pipeline route to the map
base_map.add_geojson(nord_stream_2_geojson_file_path,
                     layer_name = 'Nord Stream 2',
                     style = {"color" : "red",
                              "weight" : 3,
                             },
                     info_mode = False,
                    )
```
Here is the basemap it generates:

![image](https://user-images.githubusercontent.com/72644814/206918995-d95f78f8-ee92-49c1-8318-22ec0da5b2c0.png)

Hope that helps someone else!

**Comment by giswqs** [comment]
Here are two more examples:

- https://leafmap.org/notebooks/10_add_vector/
- https://leafmap.org/notebooks/58_bokeh/
